include/arch/arm/aarch32/misc.h

```diff
@@ -31,10 +32,35 @@ static ALWAYS_INLINE void arch_nop(void)
 	__asm__ volatile("nop");
 }
 
+#if defined(CONFIG_USERSPACE)
+#if defined(CONFIG_CPU_CORTEX_M)
+#include <arch/arm/aarch32/cortex_m/cmsis.h>
+#elif defined(CONFIG_CPU_CORTEX_R)
+#include <arch/arm/aarch32/cortex_a_r/cmsis.h>
+#endif
+
+static inline bool aarch32_thread_is_user_mode(void)
+{
+	uint32_t value;
+
+#if defined(CONFIG_CPU_CORTEX_R)
+	/*
+	 * For Cortex-R, the mode (lower 5) bits will be 0x10 for user mode.
+	 */
+	value = __get_CPSR();
+	return ((value & CPSR_M_Msk) == CPSR_M_USR) ? true : false;
+#else
+	/* return mode information */
+	value = __get_CONTROL();
+	return (value & CONTROL_nPRIV_Msk) ? true : false;
```

Comment:
Ternary op here is not necessary. The result of `value & CONTROL_nPRIV_Msk` is integer and there is an implicit integer to boolean conversion happening here.

The correct way to handle this is as follows:
```
return (value & CONTROL_nPRIV_Msk) != 0;
```

Reply:
right.  I was just moving code so I left the implementation alone assuming it was that way for a reason.  Fixed per your request.